There is a hard error in the Linux / JDK 11 build, which is a problem in the image: it is missing `git`. I guess I need to extend https://github.com/jenkinsci/jnlp-agents/pull/3.

Five test failures on the Linux / JDK 8 build. Three, including `hudson.model.LoadStatisticsTest.graph`, stem from an apparent problem in the image, affecting JFreeChart usage:

```
java.lang.UnsatisfiedLinkError: /usr/local/openjdk-8/jre/lib/amd64/libfontmanager.so: libfreetype.so.6: cannot open shared object file: No such file or directory
```

Then there are strange failures:

```
Expected: is <FAILURE>
     but: was <SUCCESS>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.jvnet.hudson.test.JenkinsRule.assertBuildStatus(JenkinsRule.java:1247)
	at hudson.tasks.ArtifactArchiverTest.archiveNotReadable(ArtifactArchiverTest.java:315)
```

and

```
java.lang.AssertionError: /foo/$ITEM_FULL_NAME should have been rejected
	at org.junit.Assert.fail(Assert.java:88)
	at jenkins.model.JenkinsBuildsAndWorkspacesDirectoriesTest.lambda$badValueForBuildsDir$3(JenkinsBuildsAndWorkspacesDirectoriesTest.java:112)
```

which I need to dig into.

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -106,6 +106,7 @@ infra.maybePublishIncrementals()
 void withMavenEnv(List envVars = [], def buildType, def javaVersion, def body) {
     if (buildType == 'Linux') {
         // I.e., a Maven container using ACI. No need to install tools.
+        sh 'apt-get update && apt-get install -y git libfontconfig1 libfreetype6' // TODO pending https://github.com/jenkinsci/jnlp-agents/pull/5
         return withEnv(envVars) {
             body.call()
         }

--- a/Jenkinsfile
+++ b/Jenkinsfile
@@ -106,7 +106,6 @@ infra.maybePublishIncrementals()
 void withMavenEnv(List envVars = [], def buildType, def javaVersion, def body) {
     if (buildType == 'Linux') {
         // I.e., a Maven container using ACI. No need to install tools.
-        sh 'apt-get update && apt-get install -y git libfontconfig1 libfreetype6' // TODO pending https://github.com/jenkinsci/jnlp-agents/pull/5
         return withEnv(envVars) {
             body.call()
         }

--- a/test/src/test/java/hudson/tasks/ArtifactArchiverTest.java
+++ b/test/src/test/java/hudson/tasks/ArtifactArchiverTest.java
@@ -54,16 +54,19 @@ import org.jenkinsci.plugins.structs.describable.DescribableModel;
 import static org.hamcrest.Matchers.lessThan;
 import static org.junit.Assert.*;
 import static org.junit.Assume.*;
+import org.junit.ClassRule;
 
 import org.junit.Rule;
 import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
 import org.jvnet.hudson.test.Issue;
 import org.jvnet.hudson.test.JenkinsRule;
 import org.jvnet.hudson.test.TestBuilder;
 import org.jvnet.hudson.test.recipes.LocalData;
 
 public class ArtifactArchiverTest {
-    
+
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
     @Rule public JenkinsRule j = new JenkinsRule();
 
     @Test
@@ -312,6 +315,7 @@ public class ArtifactArchiverTest {
         p.setAssignedNode(slave);
 
         FreeStyleBuild build = p.scheduleBuild2(0).get();
+        assumeFalse(FILENAME + " should not be readable by " + System.getProperty("user.name"), new File(build.getWorkspace().child(FILENAME).getRemote()).canRead());
         j.assertBuildStatus(Result.FAILURE, build);
         String expectedPath = build.getWorkspace().child(FILENAME).getRemote();
         j.assertLogContains("ERROR: Step ‘Archive the artifacts’ failed: java.nio.file.AccessDeniedException: " + expectedPath, build);

--- a/test/src/test/java/jenkins/model/JenkinsBuildsAndWorkspacesDirectoriesTest.java
+++ b/test/src/test/java/jenkins/model/JenkinsBuildsAndWorkspacesDirectoriesTest.java
@@ -95,16 +95,18 @@ public class JenkinsBuildsAndWorkspacesDirectoriesTest {
     @Test
     public void badValueForBuildsDir() {
         story.then((rule) -> {
-            final List<String> badValues = Arrays.asList(
+            final List<String> badValues = new ArrayList<>(Arrays.asList(
                     "blah",
                     "$JENKINS_HOME",
                     "$JENKINS_HOME/builds",
                     "$ITEM_FULL_NAME",
                     "/path/to/builds",
                     "/invalid/$JENKINS_HOME",
-                    "relative/ITEM_FULL_NAME",
-                    "/foo/$ITEM_FULL_NAME",
-                    "/$ITEM_FULLNAME");
+                    "relative/ITEM_FULL_NAME"));
+            if (!new File("/").canWrite()) {
+                badValues.add("/foo/$ITEM_FULL_NAME");
+                badValues.add("/$ITEM_FULLNAME");
+            } // else perhaps running as root
 
             for (String badValue : badValues) {
                 try {

